### PR TITLE
obs-qsv11: Check for null response on free

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description

During destruction of the QSV_Encoder_Internal the response may not be initialized and we can segfault. Add a check for this similar to the Windows implementation.

### Motivation and Context

Crashing is no good.

### How Has This Been Tested?

Currently untested, just had a user report but I don't know what the situation is that this partial encoder start happens instead of failing earlier.

### Types of changes

 - Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by kkartaltepe** [comment]
The sample stack from the user:

```
Thread 1 "obs" received signal SIGSEGV, Segmentation fault.
0x00007fff1bbd6485 in simple_free(void*, mfxFrameAllocResponse*) [clone .part.0] () from /usr/lib/x86_64-linux-gnu/obs-plugins/obs-qsv11.so
#0  0x00007fff1bbd6485 in simple_free(void*, mfxFrameAllocResponse*) [clone .part.0] () at /usr/lib/x86_64-linux-gnu/obs-plugins/obs-qsv11.so
#1  0x00007fff1bbdce37 in QSV_Encoder_Internal::ClearData() ()
    at /usr/lib/x86_64-linux-gnu/obs-plugins/obs-qsv11.so
#2  0x00007fff1bbdd077 in QSV_Encoder_Internal::~QSV_Encoder_Internal() ()
    at /usr/lib/x86_64-linux-gnu/obs-plugins/obs-qsv11.so
#3  0x00007fff1bbdb3c3 in qsv_encoder_open ()
    at /usr/lib/x86_64-linux-gnu/obs-plugins/obs-qsv11.so
#4  0x00007fff1bbd9f7a in obs_qsv_create
    (codec=codec@entry=QSV_CODEC_AVC, settings=settings@entry=0x555557339b40, encoder=encoder@entry=0x555558787dc0, useTexAlloc=useTexAlloc@entry=true)
    at ./plugins/obs-qsv11/obs-qsv11.c:872
#5  0x00007fff1bbdad83 in obs_qsv_create_tex
    (fallback_id=0x7fff1bbdfcd6 "obs_qsv11_soft_v2", encoder=0x555558787dc0, settings=0x555557339b40, codec=QSV_CODEC_AVC)
    at ./plugins/obs-qsv11/obs-qsv11.c:983
#6  obs_qsv_create_tex_h264_v2
    (settings=0x555557339b40, encoder=encoder@entry=0x555558787dc0)
    at ./plugins/obs-qsv11/obs-qsv11.c:996
#7  0x00007ffff5ed5217 in obs_encoder_initialize_internal
    (encoder=0x555558787dc0) at ./libobs/obs-encoder.c:626
#8  obs_encoder_initialize (encoder=0x555558787dc0)
    at ./libobs/obs-encoder.c:671
````

File: plugins/obs-qsv11/common_utils_linux.cpp [hunk 1]
```diff
@@ -210,6 +210,8 @@ mfxStatus simple_gethdl(mfxHDL pthis, mfxMemId mid, mfxHDL *handle)
 
 mfxStatus simple_free(mfxHDL pthis, mfxFrameAllocResponse *response)
 {
+	if (NULL == response)
```
Comment (tt2468) [inline plugins/obs-qsv11/common_utils_linux.cpp:213]: Nitpick: `if (response == nullptr)`

File: plugins/obs-qsv11/common_utils_linux.cpp [hunk 2]
```diff
@@ -210,6 +210,9 @@ mfxStatus simple_gethdl(mfxHDL pthis, mfxMemId mid, mfxHDL *handle)
 
 mfxStatus simple_free(mfxHDL pthis, mfxFrameAllocResponse *response)
 {
+	if (response == NULL)
```
Comment (Lain-B) [inline plugins/obs-qsv11/common_utils_linux.cpp:213]: Should be nullptr here probably.

**Review by Lain-B — CHANGES_REQUESTED** [review]

File: plugins/obs-qsv11/common_utils_linux.cpp (hunk 2 above)
Comment (kkartaltepe) [reply plugins/obs-qsv11/common_utils_linux.cpp]: This is what i get for copying from windows lol. Done.

**Review by tt2468 — APPROVED** [review]

**Review by RytoEX — APPROVED** [review]

**Pull request merged**